components/description-list/description-list-wrapper.js

```diff
@@ -0,0 +1,100 @@
+import { bodyCompactStylize, labelStylize } from '../typography/styles.js';
+import { css, html, LitElement } from 'lit';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const descriptionListStyles = [
+	labelStylize(css`dt`),
+	bodyCompactStylize(css`dd`),
+	css`
+		:host {
+			--d2l-dt-min-width: min-content;
+			--d2l-dt-max-width: 10rem;
+			--d2l-dd-min-width: 50%;
+		}
+		dl {
+			display: grid;
+			grid-auto-flow: row;
+			align-items: baseline;
+			grid-template-columns: minmax(var(--d2l-dt-min-width), auto) minmax(var(--d2l-dd-min-width), 1fr);
+			gap: 0.5rem 1.5rem;
+		}
+		dt {
+			max-width: var(--d2l-dt-max-width);
+		}
+		dd {
+			margin: 0;
+		}
+		d2l-dl-wrapper[stacked] dl {
+			display: block;
+		}
+		d2l-dl-wrapper[stacked] dt {
+			max-width: unset;
+		}
+		d2l-dl-wrapper[stacked] dd {
+			margin-bottom: 0.5rem;
+		}
+	`
+];
+
+/**
+ * Wraps a native <dl> element, providing styling and resize behavior.
+ * @slot - Content to wrap
+ */
+class DescriptionListWrapper extends LitElement {
+	static get properties() {
+		return {
+			stacked: { type: Boolean, reflect: true },
+			breakpoint: { type: Number, reflect: true },
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.stacked = false;
+		this.breakpoint = 350;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+		this._checkIfShouldStack();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	render() {
+		return html`
+			<slot @slotchange="${this._handleSlotChange}"></slot>
```

Comment:
Did you find that the slot change handler was necessary? I'd be surprised if there were cases where the slot changed that required a recalculation but the resize observer on the host didn't fire...

Reply:
Copied from the table wrapper - I think you're right 😄

Reply:
Table wrapper has the extra complexity in that it needs to apply CSS classes to the first/last rows and cells, so yeah it needs to listen for `slotchange` too.